### PR TITLE
[Windows] Update the Swift compiler to 5.10.1

### PR DESCRIPTION
**Pull request by shahmishal**

### CODE REVIEW

**Comment by shahmishal** [comment]
@swift-ci test Windows

**Comment by shahmishal** [comment]
@swift-ci smoke test macOS

**Comment by shahmishal** [comment]
@swift-ci smoke test Linux

**Review by etcwilde — APPROVED** [review]
Hope this fixes the arm64 6.0 build issue since 5.10.1 pulls in a fix to IRGen to handle an instruction that wasn't completely implemented in the 5.10 compiler: https://github.com/apple/swift/pull/73354

Even if it doesn't, it seems reasonable to use a release compiler for building things over the development snapshot.

**Comment by shahmishal** [comment]
@swift-ci test Windows

**Comment by etcwilde** [comment]
@swift-ci please smoke test Linux platform

**Comment by etcwilde** [comment]
@swift-ci please smoke test macOS

**Review by compnerd — APPROVED** [review]
Thanks!

File: utils/build.ps1 [hunk 1]
```diff
@@ -719,22 +719,22 @@ function Fetch-Dependencies {
 }
 
 function Get-PinnedToolchainTool() {
-  if (Test-Path "$BinaryCache\toolchains\${PinnedToolchain}\LocalApp\Programs\Swift\Toolchains\0.0.0+Asserts\usr\bin") {
-    return "$BinaryCache\toolchains\${PinnedToolchain}\LocalApp\Programs\Swift\Toolchains\0.0.0+Asserts\usr\bin"
+  if (Test-Path "$BinaryCache\toolchains\${PinnedToolchain}\LocalApp\Programs\Swift\Toolchains\5.10.1+Asserts\usr\bin") {
```
Comment (hjyamauchi) [inline utils/build.ps1:722]: It may be good to factor out this version number to a variable and set it along with the URL/SHA256 above.

**Pull request merged**